File: src/test/java/dk/cngroup/wishlist/FetchProductsTest.java
package dk.cngroup.wishlist;

import dk.cngroup.wishlist.entity.ProductRepository;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

@SpringBootTest
public class FetchProductsTest {

    @Autowired
    ProductRepository productRepository;

    //check log to see the difference in SQL executed by Hibernate
    @Test
    public void testFindProductByCode() {
        productRepository.findFirstProductByCodeIgnoreCase("randomCode");
    }
}

File: src/test/kotlin/FetchProductsTest.kt
package dk.cngroup.wishlist

import dk.cngroup.wishlist.entity.ProductRepository
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest

@SpringBootTest
class FetchProductsTest {
    @Autowired
    var productRepository: ProductRepository? = null

    //check log to see the difference in SQL executed by Hibernate
    @Test
    fun testFindProductByCode() {
        productRepository!!.findFirstProductByCodeIgnoreCase("randomCode")
    }
}
